### DISCUSSION TITLE
Python venv on top of conda base

### DISCUSSION THREAD

**Discussion by dennisvang** [pyinstaller/pyinstaller#8302] [Help]
On a Windows 10 system I use (mini)conda to manage multiple Python versions (exclusively).

Miniconda is installed per-user, so it's located in `AppData\Local\Programs`.

Now I've got a project that requires a local [`venv`][1] (for a specific reason). This venv is created using `python -m venv venv` from a terminal where the conda `base` environment has been activated.

Pyinstaller is only installed in the `venv`. The conda `base` environment only has python (and some default packages installed by conda). 

With both the conda `base` env and the `venv` activated, the project runs normally, from source.

However, an executable created using `pyinstaller.exe my-app.spec --clean --onedir` fails with import errors similar to #2084, #6564, #6565.
Specifically, I'm seeing:

>ImportError: The 'platformdirs' package is required; normally this is bundled with this package so if you get this warning, consult the packager of your distribution.
[5208] Failed to execute script 'pyi_rth_pkgres' due to unhandled exception!

However, `platformdirs` is available in `venv/Lib/site-packages/pkg_resources/_vendor`.

The following warning arises multiple times when building:

>WARNING: Assuming this is not an Anaconda environment or an additional venv/pipenv/... environment manager is being used on top, because the conda-meta folder C:\Users\me\Documents\repositories\my-project\venv\conda-meta does not exist.

Here's the pyvenv.cfg:

```
home = C:\Users\me\AppData\Local\Programs\miniconda3
include-system-site-packages = false
version = 3.11.5
executable = C:\Users\me\AppData\Local\Programs\miniconda3\python.exe
command = C:\Users\me\AppData\Local\Programs\miniconda3\python.exe -m venv C:\Users\me\Documents\repositories\my-project\venv
```

If I install all the required packages in a separate `conda` environment, i.e. without the extra python `venv`, the pyinstaller bundle does work as expected.

Is there some way to work around this issue?

[1]: https://docs.python.org/3/library/venv.html

**Comment by rokm** [comment]
Considering that the error is raised in `pyi_rth_pkgres` (run-time hook for `pkg_resources`), this should be reproducible by freezing

```python
# program.py
import pkg_resources
print("hello")
``` 

in a venv created under base miniconda environment (and considering python version is 3.11.5, that's the latest Miniconda installer/release), with just PyInstaller installed in it? Can you confirm that?

FWIW, it seems to work if I try it with the following steps:

```
(base) C:\Users\Rok>python --version
Python 3.11.5

(base) C:\Users\Rok>python -m venv venv

(base) C:\Users\Rok>venv\scripts\activate

(venv) (base) C:\Users\Rok>pip install pyinstaller

[...]

(venv) (base) C:\Users\Rok>pip freeze
altgraph==0.17.4
packaging==23.2
pefile==2023.2.7
pyinstaller==6.4.0
pyinstaller-hooks-contrib==2024.1
pywin32-ctypes==0.2.2

(venv) (base) C:\Users\Rok>pyinstaller --clean --noconfirm program.py
295 WARNING: Assuming this is not an Anaconda environment or an additional venv/pipenv/... environment manager is being used on top, because the conda-meta folder C:\Users\Rok\venv\conda-meta does not exist.
462 INFO: PyInstaller: 6.4.0, contrib hooks: 2024.1
462 INFO: Python: 3.11.5 (conda)
483 INFO: Platform: Windows-10-10.0.19045-SP0
[...]
17138 INFO: Building COLLECT COLLECT-00.toc completed successfully.

(venv) (base) C:\Users\Rok>dist\program\program.exe
hello!
```

**Reply by dennisvang** [reply]
>... should be reproducible by freezing ... Can you confirm that?

@rokm Thanks for the quick response. Yes, I would also *expect* this minimal example to reproduce the issue.

So, I followed the exact same steps from your example above, and... It works. No problem. Issue *not* reproduced.

**Reply by dennisvang** [reply]
Then I opened another powershell console, removed the venv from my original project, and created a fresh venv. Basically following the same steps as above, now with the *real* project. And... This *also* works without issue.

**Reply by dennisvang** [reply]
Now I think the issue is caused by `pyinstaller` being called while `venv` is *active* but while conda `base` is *inactive*.

It looks like I can reproduce this in my original project, but not in your minimal example. I'll look into it later.
